shared/lib/utils.py

```diff
@@ -65,13 +91,18 @@ def remove_stop_words(input_str: str, stop_words: Set[str]) -> str:
   # Example: if looking for "cat" in sentence "cat is a catty animal. i love a cat  but not cats"
   # the words "citty" and "cats" will not be matched.
   input_str = input_str.lower()
-  for words in stop_words:
-    # Using regex based replacements.
-    input_str = re.sub(rf"\b{words}\b", "", input_str)
-    # Also replace multiple spaces with a single space.
-    input_str = re.sub(r" +", " ", input_str)
 
-  # Return after removing the beginning and trailing white spaces.
+  # Protect exceptions
+  input_str = protect_exceptions(input_str, _PLACEHOLDER_MAP)
```

Comment:
nit: would `replace_exceptions_with_placeholders` and `replace_placeholders_with_exceptions` be more readable?

Reply:
done